.circleci/config.yml

```diff
@@ -383,10 +383,24 @@ jobs:
           environment:
             FOUNDRY_PROFILE: ci
           working_directory: packages/contracts-bedrock
+
+  contracts-bedrock-check-addresses:
+    docker:
+      - image: us-docker.pkg.dev/oplabs-tools-artifacts/images/ci-builder:latest
+    resource_class: large
+    steps:
+      - checkout
+      - attach_workspace: { at: "." }
+      - restore_cache:
+          name: Restore pnpm Package Cache
+          keys:
+            - pnpm-packages-v2-{{ checksum "pnpm.lock.yaml" }}
+      - check-changed:
+          patterns: sdk,contracts-bedrock,contracts
       - run:
-          name: Check addresses.json is up to date
-          command: pnpm nx build && pnpm generate:addresses:check
-          working_directory: packages/contracts-bedrock
+          name: Check generated addresses.ts file
+          command: pnpm nx run-many --target=generate:addresses:check
+    
```

Comment:
@tynes moved this to it's own build pipeline.  It blocks nothing and depends on nothing so it's what I would consider "free".

Also nx makes it so after we turn on the cache it doesn't even run unless the artifacts actually change

Reply:
Restore cache wont make it depend on js dep install?

Reply:
Correct. Restore cache depends on pnpm_lock so the js dep install only ever happens when a new dep is added or removed.  This is true for all workflows. Linking pnpm is instant and it's perfectly well cached so in general pnpm is an unproblematic bottleneck in all pipelines